homeassistant/components/airtouch4/__init__.py

```diff
@@ -0,0 +1,86 @@
+"""The AirTouch4 integration."""
+import logging
+
+from airtouch4pyapi import AirTouch
+from airtouch4pyapi.airtouch import AirTouchStatus
+
+from homeassistant.components.climate import SCAN_INTERVAL
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_HOST
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import ConfigEntryNotReady
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator, UpdateFailed
+
+from .const import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+PLATFORMS = ["climate"]
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Set up AirTouch4 from a config entry."""
+    hass.data.setdefault(DOMAIN, {})
+    host = entry.data[CONF_HOST]
+    airtouch = AirTouch(host)
+    await airtouch.UpdateInfo()
+    info = airtouch.GetAcs()
+    try:
+        if not info:
+            raise ConfigEntryNotReady
+    except OSError as error:
+        raise ConfigEntryNotReady() from error
+    coordinator = AirtouchDataUpdateCoordinator(hass, airtouch)
+    await coordinator.async_config_entry_first_refresh()
+    hass.data[DOMAIN][entry.entry_id] = {
+        "coordinator": coordinator,
+    }
```

Comment:
Nothing else is stored in `[DOMAIN][entry.entry_id]` you can set the coordinator directly here